Aztec/Classes/Extensions/String+RangeConversion.swift

```diff
@@ -241,4 +249,12 @@ public extension String {
 
         return startIndex ..< endIndex
     }
+
+    func isValidRange(_ range: Range<String.UTF16View.Index>) -> Bool {
+        isValidIndex(range.lowerBound) && isValidIndex(range.upperBound)
+    }
+
+    func isValidIndex(_ index: String.UTF16View.Index) -> Bool {
+        (self.startIndex ... self.endIndex).contains(index)
```

Comment:
```suggestion
        (self.startIndex ..< self.endIndex).contains(index)
```

Reply:
This change causes the following test to fail:

```swift
    func testThatEndIndexIsReportedAsValid() {
        let candidate = "This is a string 👍🏽"
        XCTAssertTrue(candidate.isValidIndex(candidate.endIndex))
    }
```